compiler/rustc_trait_selection/src/errors.rs

```diff
@@ -1792,6 +1793,130 @@ impl Subdiagnostic for AddPreciseCapturingAndParams {
             self.suggs,
```

Comment:
I cribbed a lot of the diagnostics logic from this subdiagnostic I previously implemented; unfortunately, not enough logic is common enough for this to be shared :/